### PR TITLE
docs() update gateways chapter (decorators, minor updates)

### PR DESCRIPTION
**Pull request by kamilmysliwiec**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: content/websockets/gateways.md [hunk 1]
```diff
@@ -54,7 +54,22 @@ handleEvent(data) {
 
 > info **Hint** `@SubscribeMessage()` and `@MessageBody()` decorators are imported from `@nestjs/websockets` package.
 
-The `handleEvent()` function takes two arguments. First one is a platform-specific [socket instance](https://socket.io/docs/server-api/#socket) and the second one is the data received from the client.
+In case you don't want to use decorators, the following code will be valid as well:
```
Comment (johnbiundo) [inline content/websockets/gateways.md]: If you would prefer not to use decorators, the following code is functionally equivalent:

File: content/websockets/gateways.md [hunk 2]
```diff
@@ -20,46 +22,70 @@ $ npm i --save @nestjs/websockets @nestjs/platform-socket.io
...
 
-In general, each gateway is listening to the same port as **HTTP server** is running on, unless your app is not a web application, or you have changed the port manually. We can change this behavior by passing an argument to the `@WebSocketGateway(80)` decorator where `80` is a chosen port number. Additionally, you can set a [namespace](https://socket.io/docs/rooms-and-namespaces/) used by this gateway with the following construction:
+In general, each gateway is listening to the same port as **HTTP server** is running on, unless your app is not a web application, or you have changed the port manually. This default behavior can be modified by passing an argument to the `@WebSocketGateway(80)` decorator where `80` is a chosen port number. You can also set a [namespace](https://socket.io/docs/rooms-and-namespaces/) used by the gateway using the following construction:
 
 ```typescript
 @WebSocketGateway(80, { namespace: 'events' })
 ```
 
-> **Warning** The gateway won't start until you put it inside the `providers` array.
+> **Warning** Gateways won't be instantiated until you put them inside the `providers` array of any existing module.
```
Comment (johnbiundo) [inline content/websockets/gateways.md]: use: Gateways aren't instantiated until the are referenced in the `providers` array of an existing module.

File: content/websockets/gateways.md [hunk 3]
```diff
@@ -20,46 +22,70 @@ $ npm i --save @nestjs/websockets @nestjs/platform-socket.io
 
 #### Overview
 
-In general, each gateway is listening to the same port as **HTTP server** is running on, unless your app is not a web application, or you have changed the port manually. We can change this behavior by passing an argument to the `@WebSocketGateway(80)` decorator where `80` is a chosen port number. Additionally, you can set a [namespace](https://socket.io/docs/rooms-and-namespaces/) used by this gateway with the following construction:
+In general, each gateway is listening to the same port as **HTTP server** is running on, unless your app is not a web application, or you have changed the port manually. This default behavior can be modified by passing an argument to the `@WebSocketGateway(80)` decorator where `80` is a chosen port number. You can also set a [namespace](https://socket.io/docs/rooms-and-namespaces/) used by the gateway using the following construction:
```
Comment (johnbiundo) [inline content/websockets/gateways.md]: use: ...listening on the same port as the **HTTP server**, unless...

File: content/websockets/gateways.md [hunk 4]
```diff
@@ -20,46 +22,70 @@ $ npm i --save @nestjs/websockets @nestjs/platform-socket.io
...
 
 ```typescript
 @WebSocketGateway(80, { namespace: 'events' })
 ```
 
-> **Warning** The gateway won't start until you put it inside the `providers` array.
+> **Warning** Gateways won't be instantiated until you put them inside the `providers` array of any existing module.
 
 The `namespace` is not a sole available option. You can pass any other property that is mentioned [here](https://socket.io/docs/server-api/). Those properties will be passed to the socket constructor during the instantiation process.
```
Comment (johnbiundo) [inline content/websockets/gateways.md]: You can pass any supported [option](https://socket.io/docs/server-api/) to the socket constructor with the second argument to the `@WebSocketGateway()` decorator, as shown below:

File: content/websockets/gateways.md [hunk 5]
```diff
@@ -20,46 +22,70 @@ $ npm i --save @nestjs/websockets @nestjs/platform-socket.io
...
 
 ```typescript
 @WebSocketGateway(81, { transports: ['websocket'] })
 ```
 
-Alright, the gateway is listening now, but we are not subscribing to the incoming messages so far. Let's create a handler that will subscribe to the `events` messages and respond to the user with the exact same data.
```
Comment (johnbiundo) [inline content/websockets/gateways.md]: Replace "Alright, the gateway is listening now, but we are not subscribing to the incoming messages so far. " with

The gateway is now listening, but we have not yet subscribed to any incoming messages.

**Comment by kamilmysliwiec** [comment]
Thank you @johnbiundo  :)

**Pull request merged**